datadog_checks_base/requirements.txt

```diff
@@ -42,3 +47,6 @@ urllib3==1.23 \
     # via requests
 uuid==1.30 \
     --hash=sha256:1f87cc004ac5120466f36c5beae48b4c48cc411968eed0eaecd3da82aa96193f
+wrapt==1.10.11 \
```

Comment:
Since it's here, I'd use it.
The decorator would just look like:
```
@wrapt.decorator
def add_tracing(wrapped, instance, args, kwargs):
```
No need for `@wraps` or any other corner case handling, `wrapt` will do it for you, see https://wrapt.readthedocs.io/en/latest/decorators.html

Reply:
I'll give that a look, thanks!